spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/MemoryReOrder.scala

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.mkldnn
+
+import breeze.linalg
+import breeze.linalg.dim
+import com.intel.analytics.bigdl.mkl.MklDnn
+import com.intel.analytics.bigdl.nn.abstractnn.{DataFormat, TensorModule}
+import com.intel.analytics.bigdl.tensor.{MklDnnTensor, Tensor}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+/**
+ * when from mkldnn layer to bigdl layer, there need to do reorder for input or gradOutput
+ */
+class MemoryReOrder(inputFormat: Int = MklDnn.MemoryFormat.any,
```

Comment:
Can we bypass the reorder? If there's a reorder in the model, throw an exception.